### PR TITLE
DEPR: Change default value for CategoricalDtype.ordered from None to False

### LINKED ISSUE #27203: CLN: Revert internal CategoricalDtype.ordered -> CategoricalDtype._ordered references
xref https://github.com/pandas-dev/pandas/pull/26403#discussion_r299144624

This will need to be done as part of the process for changing the default value of `CategoricalDtype.ordered` from `None` to `False`, which should be part of the 1.0 release.  Opening this issue to serve as a reminder.

### PR DESCRIPTION
**Pull request by jschendel**
- [X] closes #27203
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

A quick summary of the deprecation details can be found here: https://github.com/pandas-dev/pandas/pull/26403#issuecomment-507351745

### CODE REVIEW

File: pandas/core/dtypes/dtypes.py [hunk 1]
```diff
@@ -338,36 +329,45 @@ def _from_values_or_dtype(
 
         return dtype
 
+    @classmethod
+    def construct_from_string(cls, string: str):
```
Comment (jschendel) [inline pandas/core/dtypes/dtypes.py]: Can add a docstring here if needed.  For Python 3.5+, [`inspect.getdoc`](https://docs.python.org/3/library/inspect.html#inspect.getdoc) will inherit from the parent class if docstrings aren't overridden.  Basically this means that `CategoricalDtype.construct_from_string.__doc__` will be `None` but doing things like `help(CategoricalDtype.construct_from_string)` or `CategoricalDtype.construct_from_string?` will show the docstring from the parent class.  Not sure if we have other reasons for needing the `__doc__` attribute to exist though.

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -519,7 +519,7 @@ or ``matplotlib.Axes.plot``. See :ref:`plotting.formatters` for more.
 - Changed :meth:`Timedelta.resolution` to match the behavior of the standard library ``datetime.timedelta.resolution``, for the old behavior, use :meth:`Timedelta.resolution_string` (:issue:`26839`)
 - Removed previously deprecated :attr:`Timestamp.weekday_name`, :attr:`DatetimeIndex.weekday_name`, and :attr:`Series.dt.weekday_name` (:issue:`18164`)
 - Removed previously deprecated ``errors`` argument in :meth:`Timestamp.tz_localize`, :meth:`DatetimeIndex.tz_localize`, and :meth:`Series.tz_localize` (:issue:`22644`)
--
+- Changed the default value for ``ordered`` in :class:`~pandas.api.types.CategoricalDtype` from ``None`` to ``False`` (:issue:`26336`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.0.0.rst]:
```suggestion
- Changed the default value for ``ordered`` in :class:`~pandas.CategoricalDtype` from ``None`` to ``False`` (:issue:`26336`)
```

(that's the public path as in the api docs: https://dev.pandas.io/docs/reference/api/pandas.CategoricalDtype.html)

File: pandas/core/dtypes/dtypes.py [hunk 3]
```diff
@@ -562,21 +580,9 @@ def update_dtype(
...
-                # only warn if we'd actually change the existing behavior
-                msg = (
-                    "Constructing a CategoricalDtype without specifying "
-                    "`ordered` will default to `ordered=False` in a future "
-                    "version, which will cause the resulting categorical's "
-                    "`ordered` attribute to change to False; `ordered=True` "
-                    "must be explicitly passed in order to be retained"
-                )
-                warnings.warn(msg, FutureWarning, stacklevel=3)
+            new_ordered = self.ordered
```
Comment (jorisvandenbossche) [inline pandas/core/dtypes/dtypes.py]: Can you add a comment here why ordered can be None?

**Review by jorisvandenbossche — COMMENTED** [review]
I don't fully remember the previous discussion (should look back at it), but do we want to allow `ordered=None` in the main constructor? We need it internally to denote a categorical dtype that should not override the ordered property of another one, but do we need to expose this in the public constructor? (could also have a private way to created this)

If we keep it in the public constructor, we should probably document it?

File: doc/source/whatsnew/v1.0.0.rst (hunk 2 above)
Comment (jschendel) [reply doc/source/whatsnew/v1.0.0.rst]: Thanks, updated to <code>:class:\`CategoricalDtype\`</code>

File: pandas/core/dtypes/dtypes.py (hunk 3 above)
Comment (jschendel) [reply pandas/core/dtypes/dtypes.py]: There's a comment a little further up that actually addresses this (above the `new_categories` definition) but as written it's not immediately noticeable.  I've shortened the logic for both of these to be ternary statements, which hopefully makes the comment is more observable, and cleaned up the wording of the comment.

**Comment by jschendel** [comment]
> I don't fully remember the previous discussion (should look back at it), but do we want to allow ordered=None in the main constructor?

I had forgotten too but apparently my past self anticipated this and wrote a nice summary here: https://github.com/pandas-dev/pandas/pull/26403#issuecomment-507351745.

> If we keep it in the public constructor, we should probably document it?

I can add some documentation for this in a bit (after sleep + work) if we want to advertise this behavior.  There were previously some concerns that this might not be behavior we want to encourage (xref https://github.com/pandas-dev/pandas/pull/26403#discussion_r299160647) since this is in part an implementation detail to handle `"category"`, and the same behavior can be achieved by explicitly specifying `ordered`.

**Review by jreback — APPROVED** [review]
lgtm. @jorisvandenbossche

**Comment by jbrockmendel** [comment]
@jschendel needs rebase

**Comment by jschendel** [comment]
merged master and added a note regarding the usage of `ordered=None`

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
@jschendel Thanks for the update and explanation !

**Pull request merged**